File: scripts/test_inferir_tipo_veiculo.py
"""Teste inferir_tipo_veiculo — espelha app motorista."""
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parent.parent
sys.path.insert(0, str(ROOT))

from app import inferir_tipo_veiculo, imagem_checklist_parte

CASOS = [
    ('REBOQUE — EXTRA PESADO', 'pesado'),
    ('REBOQUE — CAMINHÃO', 'pesado'),
    ('REBOQUE CAMINHAO', 'pesado'),
    ('R. L. PESADO', 'pesado'),
    ('R. E. PESADO', 'pesado'),
    ('C. MEC. LEVE', 'leve'),
    ('REBOQUE LEVE', 'leve'),
    ('MOTO', 'moto'),
    ('UTILITÁRIO / VAN', 'utilitario'),
]

ok = 0
for entrada, esperado in CASOS:
    got = inferir_tipo_veiculo({'tipo': entrada, 'tipo_servico': entrada})
    status = 'OK' if got == esperado else 'FALHA'
    print(f'{status}: {entrada!r} => {got} (esperado {esperado})')
    if got == esperado:
        ok += 1

img = imagem_checklist_parte('Frente', 'pesado')
if 'pesado_frente.png' not in img:
    print(f'FALHA: imagem pesado frente => {img}')
    sys.exit(1)
print(f'OK: imagem pesado frente => {img}')

print(f'\n{ok}/{len(CASOS)} casos OK')
sys.exit(0 if ok == len(CASOS) else 1)

File: scripts/test_inferir_tipo_veiculo.mjs
/** Teste rápido inferirTipoVeiculo — node scripts/test_inferir_tipo_veiculo.mjs */
import { createRequire } from 'module';
import { pathToFileURL } from 'url';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const tsPath = path.join(__dirname, '..', 'app-motorista-novo', 'lib', 'checklist_avarias.ts');

// Duplicar lógica mínima para teste sem compilar TS
function normalizarTextoTipo(raw) {
  return String(raw || '')
    .normalize('NFD')
    .replace(/[\u0300-\u036f]/g, '')
    .toUpperCase()
    .replace(/[—–\-_/|\.]+/g, ' ')
    .replace(/\s+/g, ' ')
    .trim();
}

function inferirTipoVeiculo(...fontes) {
  const partes = fontes.filter(Boolean).map((f) => (typeof f === 'string' ? f : f.tipo || f.tipo_servico || ''));
  const t = normalizarTextoTipo(partes.join(' '));
  if (/\bEXTRA\s+PESAD|\bPESAD(O|S)\b|CAMINHAO|PRANCHAO|R\s*E\s*PES|R\s*L\s*PES|REBOQUE\s+PES/.test(t)) return 'pesado';
  if (/\bMOTO(CICLETA)?S?\b/.test(t)) return 'moto';
  if (/UTILIT|\bSUV\b|\bVAN\b|PICK\s*UP|PICKUP/.test(t)) return 'utilitario';
  if (/\bLEVE\b|C\s*MEC|AUTOMOVEL|REBOQUE\s*LEVE|\bPATIN/.test(t)) return 'leve';
  if (/\bPES\b/.test(t)) return 'pesado';
  return 'leve';
}

const casos = [
  ['REBOQUE — EXTRA PESADO', 'pesado'],
  ['R. E. PESADO', 'pesado'],
  ['C. MEC. LEVE', 'leve'],
  ['MOTO', 'moto'],
  ['MOTOCICLETA', 'moto'],
  ['UTILITÁRIO / VAN', 'utilitario'],
  ['PICK-UP', 'utilitario'],
  ['PATINS', 'leve'],
  ['REBOQUE LEVE', 'leve'],
];

let ok = 0;
for (const [entrada, esperado] of casos) {
  const got = inferirTipoVeiculo(entrada);
  const pass = got === esperado;
  console.log(`${pass ? 'OK' : 'FALHA'}: "${entrada}" => ${got} (esperado ${esperado})`);
  if (pass) ok++;
}
if (ok !== casos.length) process.exit(1);
console.log(`\n${ok}/${casos.length} casos OK`);
